pkg/controller/jobframework/reconciler.go

```diff
@@ -572,66 +576,78 @@ func (r *JobReconciler) recordAdmissionCheckUpdate(wl *kueue.Workload, job Gener
 	}
 }
 
-// IsAncestorJobManaged checks whether an ancestor job is managed by kueue.
-func (r *JobReconciler) IsAncestorJobManaged(ctx context.Context, jobObj client.Object) (bool, error) {
-	ancestor, err := r.getAncestorJobManagedByKueue(ctx, jobObj)
-	if err != nil {
-		return false, err
-	}
-	return ancestor != nil, nil
-}
-
 // getAncestorWorkload returns the Workload object of the Kueue-managed ancestor job.
-func (r *JobReconciler) getAncestorWorkload(ctx context.Context, jobObj client.Object) (*kueue.Workload, error) {
-	ancestor, err := r.getAncestorJobManagedByKueue(ctx, jobObj)
-	if err != nil || ancestor == nil {
-		return nil, err
+func (r *JobReconciler) getAncestorWorkload(ctx context.Context, ancestor client.Object) (*kueue.Workload, error) {
+	if QueueNameForObject(ancestor) == "" {
+		return nil, nil
```

Comment:
Is this check actually needed?  I suppose GetAncestorJobManagedByKueue would never return a Job without queue-name, right? but maybe this is needed for the Pod integration (instance of the TopLevelJob). If this is the case, do we have it covered in tests?

Reply:
No, it can be possible that we are returning Job without queue-name if Jobs with queue-name not found. This is need for manageJobsWithoutQueueName=true logic.

Reply:
Also foliwing original logic

https://github.com/kubernetes-sigs/kueue/blob/db2620111045588de49a2f850d79060793f344b7/pkg/controller/jobframework/reconciler.go#L295-L299

The `IsOwnerManagedByKueue(objectOwner)` returns true event if we don't have queue-name.

Reply:
Right.  If `manageJobsWithoutQueueName=true` then we need to find the highest ancestor that can be managed by Kueue, independent of whether or not it actually has a queue-name label.